features/netsocket/OnboardNetworkStack.h

```diff
@@ -121,6 +123,26 @@ class OnboardNetworkStack : public NetworkStack {
         virtual char *get_gateway(char *buf, nsapi_size_t buflen) = 0;
     };
 
+    /** Call a callback
+     *
+     *  Call a callback from the network stack context. If returns error
+     *  callback will not be called.
```

Comment:
We need a bit of text about what you can/can't do from here. Don't take significant time, and specify that of the network calls, only non-blocking UDP/datagram calls can be used.

Reply:
This is a strange API for a network stack.

Why not expose an event queue by pointer and return NULL if there isn't one used.

Then the user could choose a timer or eventqueue based on their needs.

But also +1 for mentioning that long functions could mess with the network stack.

Reply:
Actually thinking about it more this just seems like the wrong direction?

See below